src/Validator.php

```diff
@@ -44,36 +39,21 @@ class Validator
      * $conditions - array of conditions
      * $then_rules - array in $this->rules format which will be used if conditions are met
      * $else_rules - array in $this->rules format which will be used if conditions are not met
-     *
-     * @var array
      */
-    public $if_rules = [];
+    public array $if_rules = [];
 
-    /**
-     * Initialization.
-     */
     public function __construct(Model $model)
     {
-        $this->model = $model;
-
-        if (!isset($model->validator)) {
-            $model->validator = $this;
-        }
-
-        $model->onHook($model::HOOK_VALIDATE, \Closure::fromCallable([$this, 'validate']));
+        $model->onHook(Model::HOOK_VALIDATE, \Closure::fromCallable([$this, 'validate']));
```

Comment:
why model prop is no longer needed?

also make sure all you PRs have no text file without end EOL (displayed with 🚫 at the end of the diff)

Reply:
i think about this, after i tried it with php 8.1, it gives error and due to the fact that dynamic properties will be removed, so first i removed :
`$model->validator = $this;`
after that i think, why we need the model in the validator, if the model is hooked on validate and the model is passed again in that context?
so i removed even the `$model`
**I know it was public, but i think is really useless to have that unused reference**

Reply:
> after that i think, why we need the model in the validator, if the model is hooked on validate and the model is passed again in that context?

thanks, makes sense!

Reply:
the thanks goes to you, for that strict tests that make you rewrite and rethinking everything, is a good pain to feel for the project

Reply:
add https://github.com/atk4/core/pull/329/files#diff-72a46f19b75fd07f055b6fa9a5126e2dc0e3ab69f5ffa5f9f0950c22a691e086R11 trait also to the `Validator` class, then such warnings will be emit also for lower PHP versions. You should do that for every major class excl, the ones that use `DiContainerTrait` as then the trait is already included from there.